guides/cookbook.md

```diff
@@ -103,6 +103,29 @@ router.get(
 )
 ```
 
+#### Altering all responses
+
+You can also write catch-all route that will then alter all the responses. One example where this is useful would be injecting [Content Security Policy](/guides/security#section_content_security_policy__csp_).
+
+Another example is adding response headers for debugging, which is often useful if [XDN is behind another CDN](/guides/split_testing#section_third_party_cdns) or if you are troubleshooting your router rules. For example, you could respond with the value of request `x-forwarded-for` into `x-debug-xff` to see the value that XDN is receiving from the CDN:
+
+```js
+router.match({
+  path: '/:path*',
+  query: {
+    'my_site_debug': 'true',
+  }
+}, ({ setResponseHeader }) => {
+  setResponseHeader(
+    'x-debug-xff',
+    '${req:x-forwarded-for}',
+  )
+})
+  // The rest of your router...
+```
+
+The rules for interpolating the values of request and response objects can be found in the [routing](/guides/routing#section_embedded_values) guide.
+
```

Comment:
```suggestion
Note that catch-all routes that alter headers, cookies, or caching can be placed at the start of your router while allowing subsequent routes to run because they alter the request or the response without actually sending a response. See [route execution](/guides/routing#section_route_execution) for more information on route execution order and sending responses.
```